src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -112,84 +118,103 @@ private void OnDestroy()
         {
-            
-            RGDebug.LogInfo($"Creating and uploading GameplaySession on the backend, from {_startTime} to {endTime} with {_tickNumber} ticks");
-                
-            // First, create the gameplay session
-            long gameplaySessionId = -1;
-            await _rgServiceManager.CreateGameplaySession(_startTime, endTime, _tickNumber,
-                (response) =>
+
+            try
+            {
+                RGDebug.LogInfo($"Creating and uploading GameplaySession on the backend, from {startTime} to {endTime} with {tickCount} ticks");
+
+                // First, create the gameplay session
+                long gameplaySessionId = -1;
+                await RGServiceManager.GetInstance().CreateGameplaySession(startTime, endTime, tickCount,
+                    (response) =>
+                    {
+                        gameplaySessionId = response.id;
+                        RGDebug.LogInfo($"Created gameplay session with id: {response.id}");
+                    },
+                    () => { });
+
+                // If the gameplay session was not created, return
+                if (gameplaySessionId == -1)
                 {
-                    gameplaySessionId = response.id;
-                    RGDebug.LogInfo($"Created gameplay session with id: {response.id}");
-                },
-                () => {});
-            
-            // If the gameplay session was not created, return
-            if (gameplaySessionId == -1)
+                    RGDebug.LogWarning("Unable to upload gameplay session data.  Check your authorization credentials and network connection");
+                    return;
+                }
+
+                // Upload the gameplay session data
+                await RGServiceManager.GetInstance().UploadGameplaySessionData(gameplaySessionId,
+                    dataDirectoryPrefix + ".zip",
+                    () => { RGDebug.LogInfo($"Uploaded gameplay session data from {dataDirectoryPrefix}.zip"); },
+                    () => { });
+
+                // Next, upload the gameplay session screenshots
+                await RGServiceManager.GetInstance().UploadGameplaySessionScreenshots(gameplaySessionId,
+                    screenshotsDirectoryPrefix + ".zip",
+                    () => { RGDebug.LogInfo($"Uploaded gameplay session screenshots from {screenshotsDirectoryPrefix}.zip"); },
+                    () => { });
+
+                // Finally, upload the thumbnail
+                await RGServiceManager.GetInstance().UploadGameplaySessionThumbnail(gameplaySessionId,
+                    thumbnailPath,
+                    () => { RGDebug.LogInfo($"Uploaded gameplay session thumbnail from {thumbnailPath}"); },
+                    () => { });
+            }
+            catch (Exception e)
```

Comment:
Non-blocking nit - I walked through the call stack for these API calls and couldn't find a scenario where an exception would be thrown - everything would be caught and referred to the onFailure invocations (which is a spot where an exception could happen, but all of ours are empty functions). Guessing I missed something? I ended up leaving it out since I found it misleading in a lot of my failed API calls.